src/white.jl

```diff
@@ -0,0 +1,115 @@
+# white.jl
+# White and Breusch-Pagan tests for heteroskedasticity
+#
+# Copyright (C) 2020   Paul Söderlind
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+# LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+# WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+export WhiteTest, BreuschPaganTest
+
+struct WhiteTest <: HypothesisTest
+    dof::Int                 #degrees of freedom in test
+    LM::Float64              #Lagrange Multiplier test statistic, distributed as Chisq(dof)
+    testtype::Symbol
+end
+
+"""
+    WhiteTest(X, e; testtype = :White)
```

Comment:
No need to repeat "test"?
```suggestion
    WhiteTest(X, e; type = :White)
```